utils/model_wrappers/api_gateway.py

```diff
@@ -246,14 +265,37 @@ def load_chat(
             'sambanova_api_key': sambanova_api_key,
         }
         envs = {k: v for k, v in envs.items() if v is not None}
-        model = ChatSambaNovaCloud(
-            **envs,
-            model=model,
-            streaming=streaming,
-            max_tokens=max_tokens,
-            temperature=temperature,
-            top_k=top_k,
-            top_p=top_p,
-            stream_options=stream_options,
-        )
+
+        if type == 'sambastudio':
+            model = ChatSambaStudio(
+                **env,
+                model=model,
+                streaming=streaming,
+                max_tokens=max_tokens,
+                temperature=temperature,
+                top_k=top_k,
+                top_p=top_p,
+                do_sample=do_sample,
+                process_prompt=process_prompt,
+                stream_options=stream_options,
+                special_tokens=special_tokens,
+                model_kwargs=model_kwargs
+
+            )
+
+        elif type == 'sncloud':
+            model = ChatSambaNovaCloud(
```

Comment:
envs = {
            'sambanova_url': sambanova_url,
            'sambanova_api_key': sambanova_api_key,
        }
        envs = {k: v for k, v in envs.items() if v is not None}
        
        
     should inside each model conditional

Reply:
@snova-luiss